XenkoShaderExplorer/MainViewModel.cs

```diff
@@ -126,8 +127,8 @@ private IEnumerable<Shader> BuildShaderTree()
 
             foreach (var shader in shaders.Values)
             {
-                var declaration = string.Join(" ", File.ReadAllLines(shader.Path)
-                    .SkipWhile(s => !s.Trim().StartsWith("shader")) // From "shader"
+                var declaration = string.Join(" ", File.ReadLines(shader.Path)
+                    .SkipWhile(s => !s.Trim().Contains("shader ") && !s.Trim().Contains("class ")) // From "shader" or "class"
```

Comment:
You replaced `StartsWith` with `Contains` here which seems to break parsing of some shaders. For example, consider "SubsurfaceScatteringBlurShader.xksl": It starts with a long comment block containing the line "It's a simple **shader** composed of two components". By using `Contains`, we incorrectly start parsing the shader declaration at that comment line, resulting in the shader not being listed at all in the tree view.

Do you have specific examples where the old `StartsWith`-check is not sufficient?

Reply:
yes there were some shaders that had some keywords before the shader